sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DropNamespaceExec.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.analysis.NoSuchNamespaceException
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.connector.catalog.SupportsNamespaces
+
+/**
+ * Physical plan node for creating a namespace.
+ */
+case class DropNamespaceExec(
+    catalog: SupportsNamespaces,
+    namespace: Seq[String],
+    ifExists: Boolean,
+    cascade: Boolean)
+    extends V2CommandExec {
+  override protected def run(): Seq[InternalRow] = {
+    // TODO: How to handle when cascade is true?
```

Comment:
@rdblue / @cloud-fan did we discuss how we want to drop namespaces if cascade is set for v2 catalogs?

Reply:
In the v2 API, namespaces that are not empty can be rejected by throwing `IllegalStateException`. I'd say catch that if cascade is set and throw an error that cascade isn't supported.

Reply:
yea this is a good workaround now. I'm wondering if we should support cascade before 3.0, as this may break DS v2 APIs, e.g. we may need to add a cascade parameter to the `dropNamespace` API. DROP TABLE has the same problem.

Reply:
Thanks for the suggestion. I updated with a workaround solution. And +1 for adding cascade option to `dropNamespace` API. I initially tried to implement cascading option with existing APIs, but it was a bit cumbersome.